learn/guides/ballerina-openapi-support.md

```diff
@@ -0,0 +1,314 @@
+---
+layout: ballerina-openapi-support-left-nav-pages-swanlake
+title: Ballerina OpenAPI Support 
+description: Check out how the Ballerina OpenAPI tooling makes it easy for users to start developing a service documented in the OpenAPI contract.
+keywords: ballerina, programming language, openapi, open api, restful api
+permalink: /learn/ballerina-openapi-support/
+active: openapi
+intro: OpenAPI Specification is a specification that creates a RESTFUL contract for APIs, detailing all of its resources and operations in a human and machine-readable format for easy development, discovery, and integration. Ballerina OpenAPI tooling will make it easy for users to start the development of a service documented in an OpenAPI contract in Ballerina by generating Ballerina service and client skeletons. It supports the user to take the code first API design approach by generating an OpenAPI contract for the given service implementation.
+redirect_from:
+  - /learn/how-to-use-openapi-tools/
+  - /learn/how-to-use-openapi-tools
+  - /learn/using-the-openapi-tools/
+  - /learn/using-the-openapi-tools
+  - /swan-lake/learn/using-the-openapi-tools/
+  - /swan-lake/learn/using-the-openapi-tools
+  - /learn/tooling-guide/cli-tools/openapi
+  - /learn/tooling-guide/cli-tools/openapi/
+  - /learn/ballerina-openapi-support/openapi
+  - /learn/cli-documentation/openapi/#using-the-capabilities-of-the-openapi-tools/
+  - /learn/cli-documentation/openapi/#using-the-capabilities-of-the-openapi-tools
+  - /learn/cli-documentation/openapi/#openapi-validator-compiler-plugin/
+  - /learn/cli-documentation/openapi/#openapi-validator-compiler-plugin
+  - /learn/cli-documentation/openapi/
+  - /learn/cli-documentation/openapi
+---
+
+## Using the Capabilities of the OpenAPI Tools
+
+The Ballerina OpenAPI tooling support provides the following capabilities.
+ 1. Generating Ballerina service/client stubs from a given OpenAPI contract file, using the CLI tool.
+ 2. Exporting the OpenAPI definition from a given Ballerina service implementation, using the CLI tool.
+ 3. Validating the service implementation compliance with a provided OpenAPI contract, using the OpenAPI annotation.
+    - The OpenAPI compiler plugin will allow you to validate a service implementation against an OpenAPI contract during
+  the compile time.This plugin ensures that the implementation of a service does not deviate from its OpenAPI contract. 
+    
+ > Prerequisite : Download and install the latest Swan Lake Ballerina version from [here](https://ballerina.io/downloads/).
```

Comment:
```suggestion
 > Prerequisite : Download and install the latest Swan Lake Ballerina version from [here](https://ballerina.io/downloads/).
```
Shall we make this a separate section at the top before giving the instructions?